sympy/assumptions/refine.py

```diff
@@ -336,6 +336,29 @@ def refine_sign(expr, assumptions):
     return expr
 
 
+def refine_matrixelement(expr, assumptions):
+    """
+    Handler for symmetric part
+
+    Examples
+    ========
+
+    >>> from sympy.assumptions.refine import refine_matrixelement
+    >>> from sympy import Q
+    >>> from sympy.matrices.expressions.matexpr import MatrixSymbol
+    >>> X = MatrixSymbol('X', 3, 3)
+    >>> refine_matrixelement(X[0, 1], Q.symmetric(X))
+    X[0, 1]
+    >>> refine_matrixelement(X[1, 0], Q.symmetric(X))
```

Comment:
What is the output of `refine_matrixelement(X[1, 0] + X[0, 1], Q.symmetric(X))`. By intuition, I'd expect, `2*X[1, 0]` or `2*X[0, 1]`.
Can we simply use `refine(X[1, 0] + X[0, 1], Q.symmetric(X))` for getting the results(preferably the above results)?

Reply:
yes, `refine` can be used to get the results instead of `refine_matrixelement.` The function won't work for `refine_matrixelement(X[1, 0] + X[0, 1], Q.symmetric(X))`. I think this feature is not required as it wasn't mentioned in the issue.

Reply:
Though I am not sure. However the examples [here](https://docs.sympy.org/latest/modules/assumptions/refine.html) suggest that this should have happened. 
@asmeurer Should one expect such results from `refine`?

Reply:
I think its not required as the example below gives error on doing the similar thing.

> refine_abs(Abs(x),Q.negative(x))
 -x 

The above example gives the correct answer but 

> refine_abs(Abs(x)+Abs(x),Q.negative(x))
 2

which is wrong

Reply:
I think that it is an issue if `refine_abs(Abs(x)+Abs(x),Q.negative(x))` gives such value.

Reply:
@sylee957 I have raised the issue for the above example, shall I do the similar changes in `refine_matrixelement`?

Reply:
Actually, I have a second thought that the above issue like `refine_abs(Abs(x)+Abs(x),Q.negative(x))` should be closed because I think that `refine_foo` stuff are intended to use for private methods and they should always accept sanitized input.

I was confused by the comment of @czgdp1807, but I'd rather think that `refine_matrixelement(X[1, 0] + X[0, 1], Q.symmetric(X))` should not be an appropriate method for refining because the object is simply `Add` than `MatrixElement`. So you don't have to care if it works in `refine(X[1, 0] + X[0, 1], Q.symmetric(X))`.

Reply:
Ok, so I think that this request is now ready to be merged @sylee957 ?